### PR TITLE
Navigation: Add "more" toggle to subnavigation

### PR DESCRIPTION
**Pull request by gu-stav**
## What does this change?

This adds another navigation entry in the sub navigation at the end of the list, which toggles the sidebar, but opens the current pillar. This saves one click compared to the veggi-burger.

It also updates the colors of the navigation and enhances accessibility. 🎨 

## What is the value of this and can you measure success?

We'll see, but hopefully more user engangement with the navigation.

## Does this affect other platforms - Amp, Apps, etc?

No.

## Screenshots

There is a more button

screen shot 2017-04-19 at 12 49 32

which opens the corresponding section in the sidebar immediately

screen shot 2017-04-19 at 12 49 48

## Tested in CODE?

Not yet.

### CODE REVIEW

**Comment by paperboyo** [comment]
> Screenshots
> 
> No.

screenshot

File: common/app/common/NewNavigation.scala [hunk 1]
```diff
@@ -379,16 +379,14 @@ object NewNavigation {
...
       }
 
-      if (sectionList.isEmpty) {
-        "News"
-      } else {
-        sectionList.head.parentSection.name
-      }
+      sectionList.map {
+        _.parentSection.name
+      }.getOrElse("News")
```
Comment (TBonnin) [inline common/app/common/NewNavigation.scala]: It is only a matter of preference but I like to have one piece of logic per line. Something like:
```
sectionList
    .map(_.parentSection.name)
    .getOrElse("News")
```

You can even chain everything together:
```
sectionLinks
    .find(_.pageId == id)
    .map(_.parentSection.name)
    .getOrElse("News")
```

File: common/app/common/NewNavigation.scala [hunk 2]
```diff
@@ -379,16 +379,14 @@ object NewNavigation {
       }
     }
 
-    def getTopLevelSection(id: String) = {
-      val sectionList = sectionLinks.filter { item =>
+    def getParentPillar(id: String) = {
```
Comment (TBonnin) [inline common/app/common/NewNavigation.scala]: It is good practice to specify the return type for every function. The compiler is smart enough to figure it out but it is easier for developer to understand what is this function doing by only looking at the full signature.

File: common/app/common/NewNavigation.scala [hunk 3]
```diff
@@ -535,6 +533,18 @@ object NewNavigation {
       }
     }
 
+    def getActivePillar(page: Page) = {
```
Comment (TBonnin) [inline common/app/common/NewNavigation.scala]: Can you add the return type?

File: common/app/common/NewNavigation.scala [hunk 4]
```diff
@@ -535,6 +533,18 @@ object NewNavigation {
...
+      val sectionOrTagId = getSectionOrTagId(page);
+      val activeSectionLink = SectionLinks.sectionLinks.find { item =>
+        item.pageId == sectionOrTagId
+      }
+      val activePillarName = activeSectionLink.map {
+        _.parentSection.name
+      }.getOrElse("")
```
Comment (TBonnin) [inline common/app/common/NewNavigation.scala]: Is this exactly the same logic as `getParentPillar` with a different default?

File: common/app/common/NewNavigation.scala [hunk 5]
```diff
@@ -535,6 +533,18 @@ object NewNavigation {
...
+
+      (sectionOrTagId, activePillarName)
```
Comment (TBonnin) [inline common/app/common/NewNavigation.scala]: Would it make sense to create a case class `Pillar` to carry these data, rather than using a tuple?

**Review by TBonnin — CHANGES_REQUESTED** [review]
Thank you @gustavpursche 
I made a couple of comments related to the Scala code. Let me know if you want to discuss them.

File: common/app/common/NewNavigation.scala [hunk 6]
```diff
@@ -379,16 +379,19 @@ object NewNavigation {
       }
     }
 
-    def getTopLevelSection(id: String) = {
-      val sectionList = sectionLinks.filter { item =>
-        item.pageId == id
-      }
+    def getSectionLink(id: String): Option[String] = {
+      sectionLinks.find(_.pageId == id).map(_.parentSection.name)
+    }
```
Comment (TBonnin) [inline common/app/common/NewNavigation.scala]: Are `getSectionLinks` and `getSectionLink` only supposed to use internally? If yes make them `private`

File: common/app/views/fragments/nav/subNav.scala.html [hunk 7]
```diff
@@ -3,19 +3,34 @@
...
-                    ))"
-                    href="@LinkTo(link.url)" data-link-name="nav2 : subnav : @{ if(link.longTitle.isEmpty) link.title else link.longTitle}">
-                        @link.title
+                    "subnav__link" -> true,
+                    "subnav__link--current-section" -> (id == link.uniqueSection)
+                ))"
+                href="@LinkTo(link.url)"
+                data-link-name="nav2 : subnav : @{
+                    if(link.longTitle.isEmpty) link.title else link.longTitle
+                }">
```
Comment (TBonnin) [inline common/app/views/fragments/nav/subNav.scala.html]: nitpick: Keeping it on one line might make it easier to read the html source

File: common/app/common/NewNavigation.scala (hunk 6 above)
Comment (gustavpursche) [inline common/app/common/NewNavigation.scala]: Good catch.

File: common/app/mvt/MultiVariateTesting.scala [hunk 8]
```diff
@@ -39,6 +39,17 @@ object ABNewRecipeDesign extends TestDefinition(
   }
 }
 
+object ABNavigationMoreToggle extends TestDefinition(
+  name = "ab-navigation-more-toggle",
+  description = "Users in the test will see a more link in the navigation subnav",
+  owners = Seq(Owner.withGithub("gustavpursche")),
+  sellByDate = new LocalDate(2017, 4, 21)
```
Comment (gustavpursche) [inline common/app/mvt/MultiVariateTesting.scala]: You are right - talked with Stephan in the morning about it. We'll run it till Monday.

**Review by TBonnin — APPROVED** [review]

File: static/src/stylesheets/_vars.scss [hunk 9]
```diff
@@ -120,7 +120,7 @@ $paidfor-background:     $news-support-3;
 $special-report-accent:  #ffbb00;
 
 // This is only for use of the subnav in the header
-$subnav-grey: #747a81;
+$subnav-grey: #5c6268;
```
Comment (NataliaLKB) [inline static/src/stylesheets/_vars.scss]: Okay, now you have to add a screen shot! Or a gif of the more button working

**Pull request merged**